src/log.ts

```diff
@@ -92,8 +125,21 @@ class Log {
    *   const apiResponse = data[0];
    * });
    */
-  alert(entry, options, callback) {
-    this.write(Log.assignSeverityToEntries_(entry, 'ALERT'), options, callback);
+  alert(entry: Entry|Entry[], options?: WriteOptions): Promise<ApiResponse>;
+  alert(
+      entry: Entry|Entry[], options: WriteOptions,
+      callback: ApiResponseCallback): void;
+  alert(entry: Entry|Entry[], callback: ApiResponseCallback): void;
+  alert(
+      entry: Entry|Entry[],
+      optionsOrCallback?: WriteOptions|ApiResponseCallback,
+      cb?: ApiResponseCallback): void|Promise<ApiResponse> {
+    const options =
+        typeof optionsOrCallback === 'object' ? optionsOrCallback : {};
+    const callback =
+        typeof optionsOrCallback === 'function' ? optionsOrCallback : cb;
+    this.write(
+        Log.assignSeverityToEntries_(entry, 'ALERT'), options, callback!);
```

Comment:
The non-null assertion on `callback` does not seem correct. When called as the first overload, both `optionsOrCallback` and `cb` can be `undefined`. Why do you need the assertion? I suspect that the function being called has incorrect type signature. Or is this due to the magic promisification-in-the-middle that these libraries use?

Reply:
Is there an issue tracking the removal of `promisifyAll`?

A casual reader of the code will have the same question as me. This needs to be documented into a TODO in the code until this is cleaned up by the removal of `promisifyAll`.

Reply:
Sure thing.  I created a tracking issue here:
https://github.com/googleapis/google-cloud-node/issues/2887